isaric/parsers/ccp-senegal.toml

```diff
@@ -0,0 +1,739 @@
+  for.n.range = [1, 5]
+  value = { field = "daily_temp_vsorres_{n}", apply.function = "getFloat" }
+  date = { ref = "dailyDateTime" }
+
+[[observation]]
+  name = "heart_rate_bpm"
+  phase = "study"
+  for.n.range = [1, 5]
+  value = { field = "daily_hr_vsorres_{n}" }
+  date = { ref = "dailyDateTime" }
+
+[[observation]]
+  name = "respiratory_rate"
+  phase = "study"
+  for.n.range = [1, 4]
+  value = { field = "daily_rr_vsorres_{n}" }
+  date = { ref = "dailyDateTime" }
+
+[[observation]]
+  name = "systolic_blood_pressure_mmHg"
+  phase = "study"
+  for.n.range = [1, 4]
+  value = { field = "daily_sysbp_vsorres_{n}" }
+  date = { ref = "dailyDateTime" }
+
+[[observation]]
+  name = "diastolic_blood_pressure_mmHg"
+  phase = "study"
+  for.n.range = [1, 4]
+  value = { field = "daily_diabp_vsorres_{n}" }
+  date = { ref = "dailyDateTime" }
+
+[[observation]]
+  name = "oxygen_saturation_percent"
+  phase = "study"
+  for.n.range = [1, 5]
+  value = { field = "daily_oxy_vsorres_{n}" }
+  date = { ref = "dailyDateTime" }
+
+  [observation.context]
+    combinedType = "set"
+    excludeWhen = "none"
+    fields = [
+      { field = "daily_oxy_vsorresu_{n}", values = { 1 = "Room air", 2 = "Oxygen therapy" } },
+    ]
+
+[[observation]]
+  name = "mean_arterial_blood_pressure_mmHg"
+  phase = "study"
+  value = { field = "daily_meanart_vsorres_1" }
+
+  [observation.date]
+    field = "daily_vsdat"
+    source_date = "%Y-%m-%d" # makeDateTimeFromSeconds returns in ISO format
+
+    [observation.date.apply]
+      function = "makeDateTimeFromSeconds"
+      params = ["$daily_vstim_1", "%d/%m/%Y", "UTC"]
+
+[[observation]]
+  name = "heart_rate_bpm"
+  phase = "admission"
+  date = { ref = "admissionDate" }
+  value = { field = "hr_vsorres" }
+
+[[observation]]
+  name = "respiratory_rate"
+  phase = "admission"
+  date = { ref = "admissionDate" }
+  value = { field = "rr_vsorres" }
+
+[[observation]]
+  name = "systolic_blood_pressure_mmHg"
+  phase = "admission"
+  date = { ref = "admissionDate" }
+  value = { field = "sysbp_vsorres" }
+
+[[observation]]
+  name = "diastolic_blood_pressure_mmHg"
+  phase = "admission"
+  date = { ref = "admissionDate" }
+  value = { field = "admission_diabp_vsorres" }
+
+[[observation]]
+  name = "severe_dehydration"
+  phase = "admission"
+  date = { ref = "admissionDate" }
+  is_present = { field = "dehydration_vsorres", ref = "Y/N/NK" }
+
+[[observation]]
+  name = "oxygen_saturation_percent"
+  phase = "admission"
+  value = { field = "oxy_vsorres" }
+  date = { ref = "admissionDate" }
+
+  [observation.context]
+    combinedType = "set"
+    excludeWhen = "none"
+    fields = [
+      { field = "oxy_vsorresu", values = { 1 = "Room air", 2 = "Oxygen therapy" } },
+    ]
+
+[[observation]]
+  name = "cough"
+  phase = "admission"
+  start_date = { field = "cough_cestdat_v2" }
+  date = { ref = "admissionDate" }
+  duration_type = "event"
+  is_present = { field = "cough_ceoccur_v2", ref = "Y/N/NK" }
```

Comment:
Same here - make combinedType with cough sub-types

Reply:
Not sure how to combine for multiple dates

Reply:
Will use earliest date once https://github.com/globaldothealth/adtl/issues/62 is added